tests/php/ORM/DataObjectTest.php

```diff
@@ -2671,4 +2672,36 @@ public function testDBObjectEnum()
         $vals = ['25.25', '50.00', '75.00', '100.50'];
         $this->assertSame(array_combine($vals ?? [], $vals ?? []), $obj->dbObject('MyEnumWithDots')->enumValues());
     }
+
+    public function testGetDatabaseBackedField()
```

Comment:
Please also include tests for fields directly on the `$class` - such as 'Title' and 'Title.Plain' which should both return 'Title'

Please also include a couple of tests for scenarios where there's _nothing_ there - e.g. "ThisDoesntExist", "Captain.FavouriteTeam.ThisDoesntExist", and "SubTeams.ThisDoesntExist" (testing on the record itself, on a has_one, and on a many relation) which should all return `null`.

Reply:
Updated